OpenSim/Simulation/Model/ModelComponent.h

```diff
@@ -108,6 +109,38 @@ OpenSim_DECLARE_ABSTRACT_OBJECT(ModelComponent, Component);
      */
     Model& updModel();
 
+    /** Perform any computations that must occur before the Model is scaled. For
+        example, a GeometryPath must calculate and store its path length in the
+        current body position before scaling so that an owning Muscle can use
+        this information to update the properties of the muscle after scaling.
+        This method is virtual and may be implemented by any subclass of
+        ModelComponent, but all implementations must begin with a call to
+        `Super::preScale()` to execute the parent class methods before the child
+        class method. The base class implementation in ModelComponent does
+        nothing.
+        @see scale()
+        @see postScale() */
+    virtual void preScale(const SimTK::State& s, const ScaleSet& scaleSet) {};
```

Comment:
Does `preScale()` actually require a `scaleSet`? If not, we could consider removing it. I won't hold up the PR if we decide to leave this in, though.

Reply:
I think a ModelComponent developer could, in theory, want to calculate something in `preScale()` or `postScale()` that requires knowing scale factors. The original `preScale()` method took the same arguments.

Reply:
Okay; leave it in.